biogeochem/EDPhysiologyMod.F90

```diff
@@ -595,81 +645,99 @@ subroutine phenology_leafonoff(currentSite)
     type(ed_patch_type) , pointer :: currentPatch     
     type(ed_cohort_type), pointer :: currentCohort  
 
-    real(r8)           :: store_output ! the amount of the store to put into leaves -
-                                       ! is a barrier against negative storage and C starvation. 
+    real(r8) :: leaf_c                 ! leaf carbon [kg]
+    real(r8) :: store_c                ! storage carbon [kg]
+    real(r8) :: store_c_transfer_frac  ! Fraction of storage carbon used to flush leaves
+    integer  :: ipft
+    real(r8), parameter :: leaf_drop_fraction = 1.0_r8
 
     !------------------------------------------------------------------------
 
     currentPatch => CurrentSite%oldest_patch   
 
-    store_output  = 0.5_r8
-    
     do while(associated(currentPatch))    
        currentCohort => currentPatch%tallest
        do while(associated(currentCohort))        
 
-          currentCohort%leaf_litter = 0.0_r8 !zero leaf litter for today.  
-                
+          ipft = currentCohort%pft
+
+          ! Retrieve existing leaf and storage carbon
+
+          call currentCohort%prt%CheckMassConservation(ipft,0)
+
+          store_c = currentCohort%prt%GetState(store_organ, carbon12_species)
+          leaf_c  = currentCohort%prt%GetState(leaf_organ, carbon12_species)
+
           !COLD LEAF ON
-          if (EDPftvarcon_inst%season_decid(currentCohort%pft) == 1)then
+          if (EDPftvarcon_inst%season_decid(ipft) == 1)then
              if (currentSite%status == 2)then !we have just moved to leaves being on . 
                 if (currentCohort%status_coh == 1)then !Are the leaves currently off?        
-                   currentCohort%status_coh = 2    !Leaves are on, so change status to stop flow of carbon out of bstore. 
-                   if (currentCohort%laimemory <= currentCohort%bstore)then
-                      currentCohort%bl = currentCohort%laimemory !extract stored carbon to make new leaves.
+                   currentCohort%status_coh = 2    ! Leaves are on, so change status to 
+                                                   ! stop flow of carbon out of bstore. 
+                   
+                   if(store_c>nearzero) then
+                      store_c_transfer_frac = &
+                            min(EDPftvarcon_inst%phenflush_fraction(ipft)*currentCohort%laimemory, store_c)/store_c
                    else
-                      ! we can only put on as much carbon as there is in the store...
-                      ! nb. Putting all of bstore into leaves is C-starvation suicidal. 
-                      ! The tendency for this could be parameterized
-                      currentCohort%bl = currentCohort%bstore * store_output
-                   endif
-
+                      store_c_transfer_frac = 0.0_r8
```

Comment:
this seems like it might always end up being near zero? Is that right? (or does it need initializing to a default value first)

Reply:
this is weird, this might be a bug that was just created when I resolved some conflicts... nice catch

Reply:
wait, now that I look at it, i think its correct.  It is making sure that leaves will only flush if there is at least a non-zero amount of storage carbon to flush with.   Since there is a divide in line 678, we need to check to make sure the denominator is not zero, and specify a trivial clause when it is.